hoplite-core/src/main/kotlin/com/sksamuel/hoplite/FileSource.kt

```diff
@@ -47,5 +57,15 @@ sealed class FileSource {
       }.sequence()
         .mapInvalid { ConfigFailure.MultipleFailures(it) }
     }
+
+    fun fromFiles(files: List<File>): ConfigResult<List<FileSource>> {
+      return files.map { file ->
+        Try { FileInputStream(file) }.fold(
```

Comment:
Why does this try to load them first rather than error later ?

Reply:
I'm not quite sure what you mean. I tried to mirror the construction for fromPaths, but didn't really analyse exactly what was done there, so some things might not be appropriate. I re-wrote it from scratch given my understanding of what needs to be done, let me know if it looks better.